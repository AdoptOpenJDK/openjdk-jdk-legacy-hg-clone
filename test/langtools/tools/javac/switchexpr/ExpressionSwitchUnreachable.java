/*
 * @test /nodynamiccopyright/
 * @bug 8206986
 * @summary Verify reachability in switch expressions.
 * @compile/fail/ref=ExpressionSwitchUnreachable.out -XDrawDiagnostics --enable-preview -source 12 ExpressionSwitchUnreachable.java
 */

public class ExpressionSwitchUnreachable {

    public static void main(String[] args) {
        int z = 42;
        int i = switch (z) {
            case 0 -> {
                break 42;
                System.out.println("Unreachable");  //Unreachable
            }
            default -> 0;
        };
        i = switch (z) {
            case 0 -> {
                break 42;
                break 42; //Unreachable
            }
            default -> 0;
        };
        i = switch (z) {
            case 0:
                System.out.println("0");
                break 42;
                System.out.println("1");    //Unreachable
            default : break 42;
        };
        i = switch (z) {
            case 0 -> 42;
            default -> {
                break 42;
                System.out.println("Unreachable"); //Unreachable
            }
        };
        i = switch (z) {
            case 0: break 42;
            default:
                System.out.println("0");
                break 42;
                System.out.println("1");    //Unreachable
        };
        i = switch (z) {
            case 0:
            default:
                System.out.println("0");
                break 42;
                System.out.println("1");    //Unreachable
        };
    }


}
