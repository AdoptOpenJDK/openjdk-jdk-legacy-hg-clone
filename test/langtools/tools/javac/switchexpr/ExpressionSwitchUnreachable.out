ExpressionSwitchUnreachable.java:15:17: compiler.err.unreachable.stmt
ExpressionSwitchUnreachable.java:22:17: compiler.err.unreachable.stmt
ExpressionSwitchUnreachable.java:30:17: compiler.err.unreachable.stmt
ExpressionSwitchUnreachable.java:37:17: compiler.err.unreachable.stmt
ExpressionSwitchUnreachable.java:45:17: compiler.err.unreachable.stmt
ExpressionSwitchUnreachable.java:52:17: compiler.err.unreachable.stmt
- compiler.note.preview.filename: ExpressionSwitchUnreachable.java
- compiler.note.preview.recompile
6 errors